### PR TITLE
0009616: `XMLLoadFile` print error if something wrong with xml

### PR DESCRIPTION
**Pull request by CrosRoad95**
Added exclude for `Failed to open file`
Same example script will not shown any error if file not exist
[xmlErrorTest.zip](https://github.com/multitheftauto/mtasa-blue/files/1516948/xmlErrorTest.zip)
0ungdkzorfifvisi0wfywg

### CODE REVIEW

File: Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp [hunk 1]
```diff
@@ -184,6 +185,11 @@ int CLuaXMLDefs::xmlLoadFile ( lua_State* luaVM )
                             }
                         }
 
+                        xmlFile->GetLastError ( strError );
+                        const int exists = strstr ( strError, "Failed to open file" ) != NULL;
+                        if( exists==0 )
+                            argStream.SetCustomError ( strError, SString("Unable to read XML file %", strFileInput.c_str() ) );
```
Comment (Necktrox) [inline Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp]: You are missing the format specifier **s** in the format string: `"Unable to read XML file %"` should be `"Unable to read XML file %s"`

File: Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp [hunk 2]
```diff
@@ -184,6 +185,11 @@ int CLuaXMLDefs::xmlLoadFile ( lua_State* luaVM )
                             }
                         }
 
+                        xmlFile->GetLastError ( strError );
+                        const int exists = strstr ( strError, "Failed to open file" ) != NULL;
+                        if( exists==0 )
```
Comment (Necktrox) [inline Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp]: You can use `if ( FileExists ( strPath ) )` from SharedUtil

File: Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp [hunk 3]
```diff
@@ -138,6 +138,7 @@ int CLuaXMLDefs::xmlLoadFile ( lua_State* luaVM )
 
     // Grab our resource
     CLuaMain* pLuaMain = m_pLuaManager->GetVirtualMachine ( luaVM );
+    SString strError = "";
```
Comment (Necktrox) [inline Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp]: You can move this variable above the `xmlFile->GetLastError(...)` statement

File: Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp [hunk 4]
```diff
@@ -184,6 +185,11 @@ int CLuaXMLDefs::xmlLoadFile ( lua_State* luaVM )
                             }
                         }
 
+                        xmlFile->GetLastError ( strError );
```
Comment (Necktrox) [inline Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp]: `strError.empty()` could return **true**, you might want to check for it in this case.

File: Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp [hunk 5]
```diff
@@ -185,11 +184,12 @@ int CLuaXMLDefs::xmlLoadFile ( lua_State* luaVM )
                             }
                         }
 
-                        xmlFile->GetLastError ( strError );
-                        const int exists = strstr ( strError, "Failed to open file" ) != NULL;
-                        if( exists==0 )
-                            argStream.SetCustomError ( strError, SString("Unable to read XML file %s", strFileInput.c_str() ) );
-
+                        if ( FileExists ( strPath ) ) {
```
Comment (Necktrox) [inline Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp]: Put the **{** on a new line to match the code style and it should be merge-able.

Comment (CrosRoad95) [reply Shared/mods/deathmatch/logic/luadefs/CLuaXMLDefs.cpp]: done

**Pull request merged**